packages/ens-controller/src/EnsController.ts

```diff
@@ -193,6 +260,74 @@ export class EnsController extends BaseControllerV2<
     });
     return true;
   }
+
+  #getNetworkEnsSupport(network: NetworksChainId) {
+    return Boolean(ensNetworkMap[network]);
+  }
+
+  lookup(ensName: string | Promise<string>): Promise<string | null> {
+    if (!this.ethProvider) {
+      throw new Error('Provider has not been initialized.');
+    }
+    return this.ethProvider.resolveName(ensName);
+  }
+
+  reverse(address: string | Promise<string>): Promise<string | null> {
+    if (!this.ethProvider) {
+      throw new Error('Provider has not been initialized.');
+    }
+    return this.ethProvider.lookupAddress(address);
+  }
+
+  async reverseResolveAddress(nonChecksummedAddress: string) {
+    if (!this.ethProvider) {
+      throw new Error('Provider has not been initialized.');
+    }
+
+    const address = toChecksumHexAddress(nonChecksummedAddress);
+
+    if (this.state.ensResolutionsByAddress[address]) {
+      return this.state.ensResolutionsByAddress[address];
+    }
+
+    let domain: string | null;
+    try {
+      domain = await this.reverse(address);
+    } catch (error) {
+      return undefined;
```

Comment:
Nit: It might be worth preserving a debug log here.

See this module for how we handle logging from libraries: https://github.com/MetaMask/utils/blob/32d423c5e2a7bcd15b55657339617eae727adf05/src/logging.ts

Edit: Likewise for the `lookup` call below

Reply:
Sorry by "debug log" I meant using the logger I linked in this comment. I'm not sure we want to have console logs directly in controllers; typically we keep them out of libraries altogether by default. The linked logger can be enabled for diagnostics, but defaults to not logging anything. That lets the project decide whether to log or not.